### ISSUE TITLE
Add exercise topics to config

### ISSUE THREAD

**Issue by Smarticles101** [exercism/java#523]
I noticed today that none of the exercises in the config have any topics listed.  Not an urgent issue, but it might be good to add some.  

~~Should make for a super easy fix because the [xcsharp](https://github.com/exercism/xcsharp/blob/master/config.json) track already has valid topics listed in their config.  IMO this would be a good `good first patch` issue.~~

Upon further reflection, I have realized that it may not be good to just copy topics from other tracks.  Rather, I think it would be better for contributors to come up with and agree on topics for exercises.

For reference, valid topics are listed [here](https://github.com/exercism/x-common/blob/master/TOPICS.txt)

**Comment by jtigger** [comment]
Yes, @Smarticles101... will love to incorporate topics.  Yeah, and I agree that each language may trigger different topics depending on their features (that said, C# and Java are _very_ similar in feature set).  Also, I've noticed that all that can _really_ be said about an **exercise** is what kind of problem it is.  It's the **solutions** that invoke different language features — how people choose to solve them.

I am not aware of a feature that draws on this data, either.  I suspect that the redesign will directly address the gap between a series of exercises and an actual curriculum. 

For more context, digest the discussion: "Problem ordering, topics covered, and understanding of concepts required" — exercism/discussions#60.

**Comment by Smarticles101** [comment]
This is going to become relevant as soon as #704 is merged.

**Comment by FridaTveit** [comment]
We should start working on this now PR #704 is merged :)

**Comment by stkent** [comment]
Is there upstream consensus yet on how we should represent topics, @kytrinyx? https://github.com/exercism/discussions/issues/167 seems to suggest not?

**Comment by kytrinyx** [comment]
No upstream consensus yet. I'd appreciate it if you weigh in on that discussion if you have opinions.

**Comment by stkent** [comment]
Thanks for confirming; I don't have strong opinions on the mechanism used to apply labels, just wondering if we should wait on adding labels here in case the mechanism changes substantially.

**Comment by kytrinyx** [comment]
Yeah, that seems reasonable.

**Comment by stkent** [comment]
Putting this back on hold while upstream decisions get made.

**Comment by FridaTveit** [comment]
Do you think we can start working on this now @exercism/java?

**Comment by stkent** [comment]
I have seen some activity around the topics but I've not been following closely; if they are closer to canonicalized upstream and we know what the strategy is regarding tagging (tag with all topics that are unavoidable or tag with all topics that may be encountered based on implementation choices) then 👍

**Comment by FridaTveit** [comment]
I can't find any discussion abut tagging with all topics that are unavoidable or tagging with all topics that may be encountered based on implementation choices. From what I can see no one has asked that question? The only thing I did find was an explanation saying that topics should be "topics relevant to an exercise". So I would think it's topics that will most likely be relevant to the exercise even though you might be able to solve it without encountering that topic. What do you think? Do you think we can start adding topics based on that assumption (or based on something else we agree on)? Or should the question be asked upstream?

They do seem to have agreed more or less on a topics list from what I can see :)

**Comment by stkent** [comment]
> So I would think it's topics that will most likely be relevant to the exercise even though you might be able to solve it without encountering that topic.

I like this! :shipit:

**Comment by FridaTveit** [comment]
Great, I'll start looking at topics for exercises then :)

**Comment by FridaTveit** [comment]
Exercises that need topics (roughly in the order they become available):

- [ ] `rna-transcription`
- [ ] `pangram`
- [ ] `space-age`
- [ ] `acronym`

- [x] `hello-world`
- [ ] `raindrops` 
- [ ] `pig-latin` 
- [ ] `isogram` 
- [ ] `anagram` 
- [ ] `two-fer`
- [ ] `twelve-days` 
- [ ] `diamond`
- [ ] `bob` 
- [ ] `house`
- [ ] `food-chain` 
- [ ] `beer-song` 

- [ ] `hamming`
- [ ] `luhn` 
- [ ] `largest-series-product`
- [ ] `phone-number`
- [ ] `nucleotide-count`
- [ ] `series`
- [ ] `roman-numerals`

- [ ] `gigasecond`
- [ ] `allergies` 
- [ ] `meetup` 

- [ ] `scrabble-score`
- [ ] `queen-attack` 
- [ ] `minesweeper`
- [ ] `bowling`
- [ ] `word-search` 

- [ ] `difference-of-squares`
- [ ] `perfect-numbers` 
- [ ] `sum-of-multiples` 
- [ ] `sieve` 

- [ ] `secret-handshake`
- [ ] `robot-simulator` 
- [ ] `wordy` 
- [ ] `forth`

- [ ] `matrix`
- [ ] `kindergarten-garden`
- [ ] `pascals-triangle`
- [ ] `spiral-matrix`
- [ ] `transpose`

- [ ] `triangle`
- [ ] `collatz-conjecture` 
- [ ] `nth-prime` 
- [ ] `prime-factors`
- [ ] `complex-numbers`
- [ ] `pythagorean-triplet` 

- [ ] `rotational-cipher`
- [ ] `run-length-encoding` 
- [ ] `atbash-cipher` 
- [ ] `crypto-square` 
- [ ] `simple-cipher`

- [ ] `saddle-points`
- [ ] `all-your-base` 
- [ ] `clock`
- [ ] `palindrome-products` 

- [ ] `flatten-array`
- [ ] `bracket-push`
- [ ] `book-store` 
- [ ] `change` 

- [ ] `word-count`
- [ ] `etl` 
- [ ] `grade-school` 

- [ ] `robot-name`
- [ ] `ocr-numbers` 
- [ ] `rectangles` 

- [ ] `linked-list`
- [ ] `sublist`
- [ ] `simple-linked-list`
- [ ] `list-ops`
- [ ] `custom-set` 

- [ ] `binary-search`
- [ ] `binary-search-tree`

- [ ] `bank-account`

**Comment by FridaTveit** [comment]
Since there are a lot of exercises that require topics and we don't have that many `good first patch` issues at the moment, I think it would be a good idea to open `good first patch` issues for adding topics to some of the remaining exercises. If we make it some of the easier exercises but also not any of the core exercises then it shouldn't be too difficult to find topics for them while it also isn't too much of a problem if it takes a while before anyone picks them up. If we've finished adding topics to all the other exercises and no one has started doing the `good first patches` yet, then we can possibly start doing them then. Do you think that would be a good idea @exercism/java?

**Comment by stkent** [comment]
Love it!

**Comment by FridaTveit** [comment]
Great, will do that then :)

**Comment by Smarticles101** [comment]
Phew! All the exercises have topics now (that was a lot of pull requests haha). Thanks everyone who helped! 🎉
